test/linalg/dense.jl

```diff
@@ -422,6 +422,12 @@ end
         @test exp(log(A7)) ≈ A7
     end
 
+    @testset "Integer promotion tests" for (elty1, elty2) in ((Int64, Float64), (Complex{Int64}, Complex{Float64}))
```

Comment:
No big deal; but perhaps move the `for`loop to the next line such that we create one testset instead of one per loop.